subprojects/core/src/main/java/org/gradle/plugin/repository/RuleBasedPluginRepository.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugin.repository;
+
+import org.gradle.api.Action;
+import org.gradle.api.artifacts.dsl.RepositoryHandler;
+import org.gradle.plugin.repository.rules.PluginDependencyHandler;
+
+public interface RuleBasedPluginRepository extends PluginRepository {
```

Comment:
I'm quite confused by this interface. Is it the repository? It seems that it's more like a builder for a repository. What's the usage pattern here?